pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -893,13 +893,37 @@ func fetchBundle(tx *gorm.DB, req *datastore.FetchBundleRequest) (*datastore.Fet
 }
 
 // listBundles can be used to fetch all existing bundles.
-func listBundles(tx *gorm.DB) (*datastore.ListBundlesResponse, error) {
+func listBundles(tx *gorm.DB, req *datastore.ListBundlesRequest) (*datastore.ListBundlesResponse, error) {
+	if req.Pagination != nil && req.Pagination.PageSize == 0 {
+		return nil, status.Error(codes.InvalidArgument, "cannot paginate with pagesize = 0")
+	}
+
+	p := req.Pagination
+	var err error
+	if p != nil {
+		tx, err = applyPagination(p, tx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	var bundles []Bundle
 	if err := tx.Find(&bundles).Error; err != nil {
 		return nil, sqlError.Wrap(err)
 	}
 
-	resp := &datastore.ListBundlesResponse{}
+	if p != nil {
+		p.Token = ""
+		// Set token only if page size is the same than bundles len
+		if len(bundles) == int(p.PageSize) {
```

Comment:
I wonder if we should just set the token if there were any bundles? That would be consistent with the other spots that do pagination. I'd be open to changing the behavior as a separate PR, but for now this should probably be consistent?

Reply:
sure, so for now we'll force to search next until no more bundles returned

Reply:
:+1: